spec/rubocop/cop/style/return_nil_in_predicate_method_definition_spec.rb

```diff
@@ -74,6 +74,108 @@ def foo?
       RUBY
     end
 
+    it 'registers an offense when using `nil` before `rescue`' do
+      expect_offense(<<~RUBY)
+        def foo?
+          bar
+          nil
+          ^^^ Return `false` instead of `nil` in predicate methods.
+        rescue
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        def foo?
+          bar
+          false
+        rescue
+        end
+      RUBY
+    end
+
+    it 'registers an offense when using `nil` before `ensure`' do
+      expect_offense(<<~RUBY)
+        def foo?
+          bar
+          nil
+          ^^^ Return `false` instead of `nil` in predicate methods.
+        ensure
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        def foo?
+          bar
+          false
+        ensure
+        end
+      RUBY
+    end
+
+    it 'registers an offense when using `nil` inside `ensure`' do
+      expect_offense(<<~RUBY)
+        def foo?
+        ensure
+          bar
+          nil
+          ^^^ Return `false` instead of `nil` in predicate methods.
```

Comment:
I don't believe that ensure will ever return a value without an explicit return. This method would return `nil` from the empty block above.

Reply:
good catch! I'm surprised `Lint/Void` doesn't pick up on this, sounds like a false negative 👀

Reply:
Indeed! We need to improve `Lint/Void` a bit. :-)